BPA/Ch.09/ms3_multinomlogit.stan

```diff
@@ -0,0 +1,145 @@
+// -------------------------------------------------
+// States (S):
+// 1 alive at A
+// 2 alive at B
+// 3 alive at C
+// 4 dead
+// Observations (O):
+// 1 seen at A 
+// 2 seen at B
+// 3 seen at C
+// 4 not seen
+// -------------------------------------------------
+
+functions {
+  int first_capture(int[] y_i) {
+    for (k in 1:size(y_i))
+      if (y_i[k] != 4)
+        return k;
+    return 0;
+  }
+}
+
+data {
+  int<lower=0> nind;
+  int<lower=0> n_occasions;
+  int<lower=1,upper=4> y[nind, n_occasions];
+}
+
+transformed data {
+  int<lower=0,upper=n_occasions> first[nind];
+
+  for (i in 1:nind)
+    first[i] <- first_capture(y[i]);
+}
+
+parameters {
+  real<lower=0,upper=1> phiA; // Survival probability at site A
+  real<lower=0,upper=1> phiB; // Survival probability at site B
+  real<lower=0,upper=1> phiC; // Survival probability at site C
+  real<lower=0,upper=1> pA;   // Recapture probability at site A
+  real<lower=0,upper=1> pB;   // Recapture probability at site B
+  real<lower=0,upper=1> pC;   // Recapture probability at site C
+  vector[2] lpsiA;            // Logit of movement probability from site A
+  vector[2] lpsiB;            // Logit of movement probability from site B
+  vector[2] lpsiC;            // Logit of movement probability from site C
+}
+
+transformed parameters {
+  simplex[3] psiA; // Movement probability from site A
+  simplex[3] psiB; // Movement probability from site B
+  simplex[3] psiC; // Movement probability from site C
+  simplex[4] ps[4, nind, n_occasions-1];
+  simplex[4] po[4, nind, n_occasions-1];
+
+  // Constrain the transitions such that their sum is < 1
+  for (i in 1:2) {
+    psiA[i] <- exp(lpsiA[i]) / (1.0 + exp(lpsiA[1]) + exp(lpsiA[2]));
+    psiB[i] <- exp(lpsiB[i]) / (1.0 + exp(lpsiB[1]) + exp(lpsiB[2]));
+    psiC[i] <- exp(lpsiC[i]) / (1.0 + exp(lpsiC[1]) + exp(lpsiC[2]));
+  }
+  // Calculate the last transition probability
+  psiA[3] <- 1.0 - psiA[1] - psiA[2];
+  psiB[3] <- 1.0 - psiB[1] - psiB[2];
+  psiC[3] <- 1.0 - psiC[1] - psiC[2];
+
```

Comment:
Thank you for your suggestion. I incorporated the function into the model. In addition, I added comments on functions as the section C.8 of Stan reference manual recommends.